charts/nri-kube-events/templates/_helpers_compatibility.tpl

```diff
@@ -3,8 +3,351 @@ Returns a dictionary with legacy runAsUser config.
 We know that it only has "one line" but it is separated from the rest of the helpers because it is a temporary things
 that we should EOL. The EOL time of this will be marked when we GA the deprecation of Helm v2.
 */}}
-{{- define "newrelic.compatibility.securityContext.pod" -}}
-{{- if  .Values.runAsUser -}}
+{{- define "nri-kube-events.compatibility.securityContext.pod" -}}
+{{- if .Values.runAsUser -}}
 runAsUser: {{ .Values.runAsUser }}
 {{- end -}}
 {{- end -}}
+
+
+
+
+{{- /* Functions to fetch integration and agent image configurations from the old .Values.image */ -}}
+{{- define "nri-kube-events.compatibility.old.integration.registry" -}}
+    {{- if .Values.image -}}
```

Comment:
I am bit scared by the verbosity, I know that this is due the "dig" command not being available

Could we manually merge the values and then use the image function of the common library?

```
{{- define "nri-kube-events.compatibility.old.integration.image" -}}
    {{ $tmp := .images.integration}}

    {{- if .Values.image -}}
        {{- if .Values.image.kubeEvents.tag -}}
                {{ $tmp.tag := .Values.image.kubeEvents.tag }}.  //pseudocode 😓 
        {{- end -}}
    {{- end -}}
[...]

{{ include "newrelic.common.images.image" ( dict "imageRoot"  $tmp "context" .) }}

{{- end -}}
```

Reply:
we start having the "new" struct and we override it with whatever old value if available, then we call the classical function

Reply:
I know that the verbosity and the amount of helpers are overwhelming, but all of them do the same:
```mustache
    {{- if .Values.images -}}
        {{- if .Values.images.agent -}}
            {{- if .Values.images.agent.pullPolicy -}}
                {{ .Values.images.agent.pullPolicy }}
            {{- end -}}
        {{- end -}}
    {{- end -}}
```
And in the end, it makes really easy to read:
```mustache
{{- $globalRegistry := include "nri-kube-events.compatibility.global.registry" . -}}
{{- $oldRegistry := include "nri-kube-events.compatibility.old.agent.registry" . -}}
{{- $newRegistry := include "nri-kube-events.compatibility.new.agent.registry" . -}}
{{- $registry := $oldRegistry | default $newRegistry | default $globalRegistry -}}
```

If we start using functions from the common library, we should start to have into account that the only globals are `registry` and `pullSecrets` and we could end making the code harder to read because of messing the precedence of the common library with the precedence needed for this chart.

Think also that this helper will EOL in a few months. Everything in the `_helpers_compatibility.tpl` is a workaround.